File: app/src/main/java/wxm/KeepAccount/ui/utility/HelperDayNotesInfo.java
package wxm.KeepAccount.ui.utility;

import android.content.Context;
import android.content.res.Resources;
import android.os.Build;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import java.util.Objects;

import wxm.KeepAccount.R;
import wxm.KeepAccount.utility.ContextUtil;
import wxm.androidutil.ViewHolder.ViewHolder;

/**
 * day data helper
 * Created by WangXM on 2017/1/22.
 */
public class HelperDayNotesInfo {
    private static int CR_PAY;
    private static int CR_INCOME;
    private static int DIM_FULL_WIDTH;

    static {
        Context ct = Objects.requireNonNull(ContextUtil.Companion.getInstance());
        Resources res = ct.getResources();
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            Resources.Theme te = ct.getTheme();
            CR_PAY = res.getColor(R.color.darkred, te);
            CR_INCOME = res.getColor(R.color.darkslategrey, te);
        } else {
            CR_PAY = res.getColor(R.color.darkred);
            CR_INCOME = res.getColor(R.color.darkslategrey);
        }

        DIM_FULL_WIDTH = (int) res.getDimension(R.dimen.rl_amount_info_width);
    }

    /**
     * fill note UI
     *
     * @param vh            holder for view
     * @param pay_count     pay count in one day
     * @param pay_amount    pay amount in one day
     * @param income_count  income count in one day
     * @param income_amount income amount in one day
     * @param amount        balance amount in one day
     */
    public static void fillNoteInfo(ViewHolder vh, String pay_count, String pay_amount,
                                    String income_count, String income_amount, String amount) {
        boolean b_pay = !"0".equals(pay_count);
        boolean b_income = !"0".equals(income_count);

        RelativeLayout rl_p = vh.getView(R.id.rl_pay);
        RelativeLayout rl_i = vh.getView(R.id.rl_income);

        rl_i.setVisibility(View.VISIBLE);
        rl_p.setVisibility(View.VISIBLE);

        ImageView i_iv = vh.getView(R.id.iv_income_line);
        ImageView p_iv = vh.getView(R.id.iv_pay_line);
        if (b_pay) {
            ViewGroup.LayoutParams i_para = i_iv.getLayoutParams();
            i_para.width = DIM_FULL_WIDTH;
            i_iv.setLayoutParams(i_para);

            TextView tv = vh.getView(R.id.tv_pay_count);
            tv.setText(pay_count);

            tv = vh.getView(R.id.tv_pay_amount);
            tv.setText(pay_amount);
        } else {
            rl_p.setVisibility(View.GONE);
        }

        if (b_income) {
            ViewGroup.LayoutParams p_para = p_iv.getLayoutParams();
            p_para.width = DIM_FULL_WIDTH;
            p_iv.setLayoutParams(p_para);

            TextView tv = vh.getView(R.id.tv_income_count);
            tv.setText(income_count);

            tv = vh.getView(R.id.tv_income_amount);
            tv.setText(income_amount);
        } else {
            rl_i.setVisibility(View.GONE);
        }

        if (b_income && b_pay) {
            float pay = Float.valueOf(pay_amount);
            float income = Float.valueOf(income_amount);
            ImageView iv = vh.getView(pay < income ? R.id.iv_pay_line : R.id.iv_income_line);
            ViewGroup.LayoutParams para = iv.getLayoutParams();
            float ratio = (pay > income ? income : pay)
                    / (pay < income ? income : pay);
            //Log.v(LOG_TAG, "ratio : " + ratio + ", width : " + org_para.width);

            para.width = (int) (DIM_FULL_WIDTH * ratio);
            iv.setLayoutParams(para);
        }

        TextView tv = vh.getView(R.id.tv_amount);
        tv.setText(amount);
        tv.setTextColor(amount.startsWith("+") ? CR_INCOME : CR_PAY);
    }
}

File: app/src/main/java/wxm/KeepAccount/ui/utility/HelperDayNotesInfo.kt
package wxm.KeepAccount.ui.utility

import android.view.View
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView

import wxm.KeepAccount.R
import wxm.KeepAccount.utility.ContextUtil
import wxm.androidutil.app.AppBase
import wxm.androidutil.ui.view.ViewHolder

/**
 * day data helper
 * Created by WangXM on 2017/1/22.
 */
object HelperDayNotesInfo {
    private val CR_PAY: Int = AppBase.getColor(R.color.darkred)
    private val CR_INCOME: Int = AppBase.getColor(R.color.darkslategrey)
    private val DIM_FULL_WIDTH: Int = AppBase.getDimension(R.dimen.rl_amount_info_width).toInt()

    /**
     * fill note UI
     *
     * @param vh            holder for view
     * @param pay_count     pay count in one day
     * @param pay_amount    pay amount in one day
     * @param income_count  income count in one day
     * @param income_amount income amount in one day
     * @param amount        balance amount in one day
     */
    fun fillNoteInfo(vh: ViewHolder, pay_count: String, pay_amount: String,
                     income_count: String, income_amount: String, amount: String) {
        val valPay = "0" != pay_count
        val valIncome = "0" != income_count

        val rlPay = vh.getView<RelativeLayout>(R.id.rl_pay)
        val rlIncome = vh.getView<RelativeLayout>(R.id.rl_income)

        rlIncome.visibility = View.VISIBLE
        rlPay.visibility = View.VISIBLE

        val lineIncome = vh.getView<ImageView>(R.id.iv_income_line)
        val linePay = vh.getView<ImageView>(R.id.iv_pay_line)
        if (valPay) {
            val paramIncome = lineIncome.layoutParams
            paramIncome.width = DIM_FULL_WIDTH
            lineIncome.layoutParams = paramIncome

            var tv = vh.getView<TextView>(R.id.tv_pay_count)
            tv.text = pay_count

            tv = vh.getView(R.id.tv_pay_amount)
            tv.text = pay_amount
        } else {
            rlPay.visibility = View.GONE
        }

        if (valIncome) {
            val paramPay = linePay.layoutParams
            paramPay.width = DIM_FULL_WIDTH
            linePay.layoutParams = paramPay

            var tv = vh.getView<TextView>(R.id.tv_income_count)
            tv.text = income_count

            tv = vh.getView(R.id.tv_income_amount)
            tv.text = income_amount
        } else {
            rlIncome.visibility = View.GONE
        }

        if (valIncome && valPay) {
            val pay = java.lang.Float.valueOf(pay_amount)!!
            val income = java.lang.Float.valueOf(income_amount)!!
            val iv = vh.getView<ImageView>(if (pay < income) R.id.iv_pay_line else R.id.iv_income_line)
            val para = iv.layoutParams
            val ratio = (if (pay > income) income else pay) / if (pay < income) income else pay
            //TagLog.v(LOG_TAG, "ratio : " + ratio + ", width : " + org_para.width);

            para.width = (DIM_FULL_WIDTH * ratio).toInt()
            iv.layoutParams = para
        }

        val tv = vh.getView<TextView>(R.id.tv_amount)
        tv.text = amount
        tv.setTextColor(if (amount.startsWith("+")) CR_INCOME else CR_PAY)
    }
}
